### PR TITLE
Eslint changes

### PR DESCRIPTION
**Pull request by ericcater**
- created shared eslint config for backend projects
- added complexity rule
- added no-return-await rule
- added no import from @odst/types in frontend
- added no return await rule
- added explicit return type (or servicer/resolver files)
- added no misused promises rule

### CODE REVIEW

File: .eslintrc.api.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          { "checksVoidReturn": false }
+        ]
+      }
+    },
+    {
+      "files": ["*.resolver.ts", "*.service.ts"],
+      "parser": "@typescript-eslint/parser",
+      "plugins": ["@typescript-eslint"],
+      "rules": {
+        //TODO would like to make up this to error eventually
```
Comment (yurisim) [inline .eslintrc.api.json:26]: I thought comments weren't permitted in JSON?

File: .eslintrc.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
               }
             ]
           }
-        ]
+        ],
+        "complexity": ["warn", 4]
```
Comment (yurisim) [inline .eslintrc.json:22]: We should tell the team about this.

**Review by yurisim — APPROVED** [review]
Merge after telling the team about the Complexity changes

File: .eslintrc.api.json (hunk 1 above)
Comment (ericcater) [reply .eslintrc.api.json:26]: I do what I want.

**Review by Mikehickey7 — DISMISSED** [review]
Looks good boo

**Review by yurisim — APPROVED** [review]

**Review by Mikehickey7 — APPROVED** [review]

**Review by yurisim — APPROVED** [review]

**Pull request merged**